File: code/smv-debugger/src/main/java/org/fbme/smvDebugger/panel/items/GlobalTime.java
package org.fbme.smvDebugger.panel.items;

import org.fbme.smvDebugger.panel.mvc.DebugPanelMVCItem;
import org.fbme.smvDebugger.panel.mvc.DebugPanelModel;
import org.fbme.smvDebugger.panel.mvc.GlobalTimeModel;

import javax.swing.*;

public class GlobalTime extends JLabel implements DebugPanelMVCItem {
    private static final String LEFT_PADDING = "     ";
    private DebugPanelModel model;

    @Override
    public void setPanelModel(final DebugPanelModel model) {
        this.model = model;
    }

    @Override
    public void initView() {
    }

    @Override
    public void initController() {
        model.getGlobalTimeModel().addPropertyChangeListener(GlobalTimeModel.GLOBAL_TIME, event -> {
            final String time = (String) event.getNewValue();
            setText(LEFT_PADDING + time);
        });
    }
}

File: code/smv-debugger/src/main/kotlin/org/fbme/smvDebugger/panel/items/GlobalTime.kt
package org.fbme.smvDebugger.panel.items

import org.fbme.smvDebugger.panel.mvc.DebugPanelMVCItem
import org.fbme.smvDebugger.panel.mvc.DebugPanelModel
import org.fbme.smvDebugger.panel.mvc.GlobalTimeModel
import java.beans.PropertyChangeEvent
import javax.swing.JLabel

class GlobalTime : JLabel(), DebugPanelMVCItem {
    private var model: DebugPanelModel? = null
    override fun setPanelModel(model: DebugPanelModel) {
        this.model = model
    }

    override fun initView() {}
    override fun initController() {
        model!!.globalTimeModel
            .addPropertyChangeListener(GlobalTimeModel.GLOBAL_TIME) { event: PropertyChangeEvent ->
                val time = event.newValue as String
                text = LEFT_PADDING + time
            }
    }

    companion object {
        private const val LEFT_PADDING = "     "
    }
}
